resolver/resolver_test.go

```diff
@@ -0,0 +1,118 @@
+package resolver
+
+import (
+	"net"
+	"net/http"
+	"testing"
+)
+
+func TestResolve8888(t *testing.T) {
+	r, err := NewResolver([]string{"8.8.8.8:53"})
+	if err != nil {
+		t.FailNow()
+	}
+
+	net.DefaultResolver = r
+
+	resp, err := http.Get("https://www.google.com/")
+
+	if err != nil {
+		t.Logf("error from http.Get(): %s", err)
+		t.FailNow()
+	}
+
+	resp.Body.Close()
+}
+
+func TestResolveAddresses(t *testing.T) {
+	table := map[string]struct {
+		input          []string
+		want           []string
+		expectError    bool
+		expectMismatch bool
+	}{
+		"Good list": {
+			input: []string{
+				"8.8.8.8:53",
+				"9.9.9.9:1234",
+				"2.3.4.5",
+			},
+			want: []string{
+				"8.8.8.8:53",
+				"9.9.9.9:1234",
+				"2.3.4.5:53",
+			},
+			expectError:    false,
+			expectMismatch: false,
+		},
+		"Mismatch list": {
+			input: []string{
+				"9.9.9.9:1234",
+			},
+			want: []string{
+				"9.9.9.9:53",
+			},
+			expectError:    false,
+			expectMismatch: true,
+		},
+		"Parse error list": {
+			input: []string{
+				"abcd.com:53",
+			},
+			expectError: true,
+		},
+	}
+	for subtest, tdata := range table {
+		t.Run(subtest, func(t *testing.T) {
```

Comment:
Can these sub tests be ran in `t.Parallel()`?

Reply:
why?

Reply:
Reduced test run-time.